### ISSUE TITLE
The used list of keywords is from kotlin but should be for typescript

### ISSUE THREAD

**Issue by Legion2** [outfoxx/typescriptpoet#23]
The typescript builder checks names for keywords, but it uses a list of koltin keywords instead of a list of typescript keywords:

https://github.com/outfoxx/typescriptpoet/blob/62d7d4b276b6039bd04729ddb135a9428e61b03e/src/main/java/io/outfoxx/typescriptpoet/Utils.kt#L158-L188

The list is missing they keyword `enum` as an example (which is also a keyword in kotlin).

**Comment by kdubb** [comment]
You are correct! Both the Swift and TypeScript poets originated from meddling with the Kotlin & Java Poets; there are deficiencies.

Welcome any PRs or information you can contribute. Specifically, where the TS keyword list is defined would be very helpful.

**Comment by Legion2** [comment]
I found this in the official typescript parser source code: https://github.com/microsoft/TypeScript/blob/b1bec34f20690c6fdf6cc6e3d8cacee1e075f5ae/src/compiler/types.ts#L141
